### PR TITLE
Initial code for Azure Monitor Exporters

### PR DESCRIPTION
**Pull request by lzchen**
Most of the code is taken from:
https://github.com/open-telemetry/opentelemetry-python/tree/master/ext/opentelemetry-ext-azure-monitor

Next Steps:
1. Change relevant files to Microsoft specific
2. Tests
3. CI
4. Upload new PyPi Package
5. Remove previous PyPi Package

### CODE REVIEW

File: azure_monitor/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[metadata]
+name = opentelemetry-azure-monitor-exporters-python
+description = Azure Monitor exporters using OpenTelemetry
+long_description = file: README.rst
+long_description_content_type = text/x-rst
+author = Microsoft
+author_email = appinsightssdk@microsoft.com
```
Comment (hectorhdzg) [inline azure_monitor/setup.cfg]: TelReachSDK@microsoft.com?

Comment (reyang) [reply azure_monitor/setup.cfg]: That name doesn't make sense.

File: azure_monitor/README.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+===========================================
+
+This library provides integration with Azure Monitor using OpenTelemetry.
+
+Installation
+------------
+
+::
+
+    pip install azure-monitor-opentelemetry-exporters-python
```
Comment (hectorhdzg) [inline azure_monitor/README.rst]: The package is called differently in setup.cfg file
opentelemetry-azure-monitor-exporters-python

File: azure_monitor/README.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+References
+----------
+
+* `Azure Monitor <https://docs.microsoft.com/azure/azure-monitor/>`_
+* `OpenTelemetry Project <https://opentelemetry.io/>`_
```
Comment (hectorhdzg) [inline azure_monitor/README.rst]: https://github.com/open-telemetry/opentelemetry-python would be better here, saving some clicks and confusion for people using this

File: azure_monitor/examples/client.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import requests
+
+from azure_monitor import AzureMonitorSpanExporter
```
Comment (hectorhdzg) [inline azure_monitor/examples/client.py]: Do we really need azure_monitor folder?, feels redundant

File: azure_monitor/setup.cfg (hunk 1 above)
Comment (hectorhdzg) [reply azure_monitor/setup.cfg]: Well appinsightssdk@microsoft.com is actually an alias for node.js not python, maybe updating the alias to actually include all AppInsights SDK people would be nice

**Review by reyang — CHANGES_REQUESTED** [review]
Given the majority of the code was copied from OpenTelemetry, we should keep the LICENSE info - otherwise we are against the OSS license.

File: azure_monitor/README.rst (hunk 2 above)
Comment (reyang) [reply azure_monitor/README.rst]: Why would one want to name a python package with `python` in the name? Do we have an npm package named `foobar-javascript` or `foobar-ecmascript`?

Comment (reyang) [inline azure_monitor/README.rst]: This name is very weird.

File: azure_monitor/README.rst [hunk 5]
```diff
@@ -0,0 +1,17 @@
+Azure Monitor Exporters using OpenTelemetry
```
Comment (reyang) [inline azure_monitor/README.rst]: What does `using` mean?

File: azure_monitor/examples/server.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import flask
+import requests
+
+from opentelemetry import trace
+from azure_monitor import AzureMonitorSpanExporter
```
Comment (reyang) [inline azure_monitor/examples/server.py]: This is not following the style guideline.

File: azure_monitor/setup.cfg [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+python_requires = >=3.4
+package_dir=
+    =src
+packages=find_namespace:
+install_requires =
+    opentelemetry-api
+    opentelemetry-sdk
+
+[options.packages.find]
+where = src
```
Comment (reyang) [inline azure_monitor/setup.cfg:46]: This is against github rule.

File: azure_monitor/src/azure_monitor/__init__.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+from azure_monitor.trace import AzureMonitorSpanExporter
+from azure_monitor.version import __version__
+
+__all__ = ["AzureMonitorSpanExporter"]
```
Comment (reyang) [inline azure_monitor/src/azure_monitor/__init__.py]: same here and other places.

**Comment by reyang** [comment]
> Most of the code is taken from

It is very hard to tell which part are taken and which part are modified.
Please split this into two PRs - one simply taking, another one making the modifications.

File: azure_monitor/README.rst (hunk 2 above)
Comment (hectorhdzg) [reply azure_monitor/README.rst]: Funny you mention that, we actually do have foobar-js type of names in npm, but that is a different story, I would suggest opentelemetry-azure-monitor-exporter as package name, also we need to cleanup existing one in pypi https://pypi.org/project/opentelemetry-ext-azure-monitor/

Comment (reyang) [reply azure_monitor/README.rst]: +1 on the need to clean up existing one if we decided to go with a different name.

File: azure_monitor/setup.cfg (hunk 1 above)
Comment (lzchen) [reply azure_monitor/setup.cfg]: We will keep appinsightssdk@microsoft.com and update the alias to include our team.

File: azure_monitor/README.rst (hunk 2 above)
Comment (lzchen) [reply azure_monitor/README.rst]: `opentelemetry-azure-monitor-exporter` sounds good. Will have to clean up the existing PyPi package.

File: azure_monitor/examples/client.py (hunk 4 above)
Comment (lzchen) [reply azure_monitor/examples/client.py]: For future exporters to different ingestion endpoints, they can live in separate folders.

File: azure_monitor/README.rst (hunk 2 above)
Comment (lzchen) [reply azure_monitor/README.rst]: Renamed to `opentelemetry-azure-monitor-exporter`

File: azure_monitor/examples/client.py (hunk 4 above)
Comment (hectorhdzg) [reply azure_monitor/examples/client.py]: Now that the repo is called differently I totally agree

File: azure_monitor/README.rst (hunk 5 above)
Comment (lzchen) [reply azure_monitor/README.rst]: Means has dependencies on.

File: azure_monitor/setup.cfg (hunk 7 above)
Comment (lzchen) [reply azure_monitor/setup.cfg:46]: Can you explain what you mean?

File: azure_monitor/src/azure_monitor/__init__.py (hunk 8 above)
Comment (lzchen) [reply azure_monitor/src/azure_monitor/__init__.py]: What does that mean?

File: azure_monitor/setup.cfg (hunk 7 above)
Comment (reyang) [reply azure_monitor/setup.cfg:46]: image

**Comment by lzchen** [comment]
@reyang  @hectorhdzg 
I've reverted the changes and did a complete copy of the exporter from OT.
I will take note of the comments here and apply them to a different PR.

**Review by hectorhdzg — APPROVED** [review]

**Pull request merged**